### DISCUSSION TITLE
xDrip-formula for calculating the "trend arrow"

### DISCUSSION THREAD

**Discussion by emp-00** [NightscoutFoundation/xDrip#2394] [Q&A]
Can anybody point me please with a link to the xDrip+ code section which calculates the "trend arrows", see screenshot?

![image](https://user-images.githubusercontent.com/60714349/197388284-d67686ba-e6d5-405b-9d47-b5474a40a099.png)

Alternatively, can you describe the xDrip formula, e.g. `calculate the average of the last 3 (?) bg values` and `decide based on this case matrix`:
```
>10 --> double arrow up
>5 --> single arrow up
>3 --> 45 degree arrow up
< 3 --> 45 degree arrow down
...
else --> flat arrow
```

Background info: I'm thinking about implementing the arrow myself in a customized gadget because of an [apparent xDrip-bug](https://github.com/NightscoutFoundation/xDrip/discussions/2372) in connection with the Nightscout API "trend arrow" output.

Many thanks!

**Comment by Navid200** [comment]
I believe the trend arrow represents the slope of the line connecting the current reading to the previous reading.
